contrib/awslambda/python/src/python/pants/contrib/awslambda/python/tasks/lambdex_run.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import functools
+import os
+import shutil
+
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.task.task import Task
+from pants.util.contextutil import temporary_dir
+from pants.util.dirutil import safe_mkdir_for
+from pants.util.fileutil import atomic_copy
+
+from pants.contrib.awslambda.python.targets.python_awslambda import PythonAWSLambda
+from pants.contrib.awslambda.python.tasks.lambdex_prep import LambdexPrep
+
+
+class LambdexRun(Task):
+  """Runs Lambdex to convert pexes to AWS Lambda functions.
+
+  Note that your pex must be built to run on Amazon Linux, e.g., if it contains native code.
+  When deploying the lambda, its handler should be set to `lambdex_handler.handler`, which
+  is a wrapper around the target-specified handler.
+  """
+
+  @staticmethod
+  def _is_python_lambda(target):
+    return isinstance(target, PythonAWSLambda)
+
+  @classmethod
+  def product_types(cls):
+    return ['python_aws_lambda']
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(LambdexRun, cls).prepare(options, round_manager)
+    round_manager.require_data(LambdexPrep.tool_instance_cls)
+    round_manager.require('pex_archives')
+
+  @classmethod
+  def create_target_dirs(self):
+    return True
+
+  def execute(self):
+    targets = self.get_targets(self._is_python_lambda)
+    with self.invalidated(targets=targets) as invalidation_check:
+      python_lambda_product = self.context.products.get_data('python_aws_lambda', dict)
+      for vt in invalidation_check.all_vts:
```

Comment:
I'm not sure what `vt` refers to. Disclaimer I've never used AWS Lambdas.

Reply:
VersionedTarget

https://github.com/pantsbuild/pants/blob/545a00c2bfc71bc08d453c8b4151413a63f8aa38/src/python/pants/invalidation/cache_manager.py#L170

It's a common pattern for a lot of the tasks.

Reply:
`VersionedTarget` -- it's a common abbreviation used in implementations of v1 tasks.